scripts/linters/pylint_extensions_test.py

```diff
@@ -2915,3 +2915,164 @@ def test_usage_of_is_not_on_none_does_not_add_message(self):
         compare_node = if_node.test
         with self.checker_test_object.assertNoMessages():
             self.checker_test_object.checker.visit_compare(compare_node)
+
+
+class DisallowedFunctionsCheckerTests(unittest.TestCase):
+    """Unit tests for DisallowedFunctionsChecker
+    """
+
+    def setUp(self):
+        super(DisallowedFunctionsCheckerTests, self).setUp()
+        self.checker_test_object = testutils.CheckerTestCase()
+        self.checker_test_object.CHECKER_CLASS = (
+            pylint_extensions.DisallowedFunctionsChecker)
+        self.checker_test_object.setup_method()
+
+    def test_disallowed_removals(self):
+        self.checker_test_object.checker.config.disallowed_functions = [
+            b'example_func',
+        ]
+
+        call1 = astroid.extract_node(
+            """
+        example_func() #@
+        """)
+
+        message_remove_example = testutils.Message(
+            msg_id='remove-disallowed-function-calls',
+            node=call1,
+            args=b'example_func'
+        )
+
+        with self.checker_test_object.assertAddsMessages(
+            message_remove_example):
+            self.checker_test_object.checker.visit_call(call1)
+
+    def test_disallowed_replacements(self):
+        self.checker_test_object.checker.config.disallowed_functions = [
+            b'now=>datetime.datetime.utcnow',
+            b'assertEquals=>self.assertEqual',
+            b'StringIO=>python_utils.string_io',
+            b'urlsplit=>python_utils.url_split',
+            b'urlparse=>python_utils.url_parse',
+            b'urlunsplit=>python_utils.url_unsplit',
+            b'parse_qs=>python_utils.parse_query_string',
+            b'unquote=>python_utils.urllib_unquote',
+            b'urljoin=>python_utils.url_join',
+            b'next=>python_utils.NEXT',
+            b'range=>python_utils.RANGE',
+            b'round=>python_utils.ROUND',
+            b'str=>python_utils.convert_to_bytes or python_utils.UNICODE',
+            b'zip=>python_utils.ZIP',
+            b'basestring=>python_utils.BASESTRING',
+            b'iteritems=>items',
+            b'itervalues=>values',
+            b'iterkeys=>keys',
+        ]
```

Comment:
[optional] We don't have to test each of the functions written in the `.pylintrc` (we can assume .pylintrc to be independent). Maybe just have a few cases here?

Reply:
Done, reduced to 4 cases.